### DISCUSSION TITLE
Two comments on cco:Plan

### DISCUSSION THREAD

**Discussion by avsculley** [CommonCoreOntology/CommonCoreOntologies#343] [Announcements]
Hello,

I have been working on an ontology of deviant acts using CCO. I wanted to bring two comments to your attention, which came up in my work on deviant acts. Both comments concern cco:Plan.

1. Some Agent need not expect to achieve some Objective.
(a) One reason is that expectations require mental capabilities, and some agents do not have mental capabilities. Software agents, for example, lack mental capabilities, but still achieve some Objective as the only agents in a planned process.
(b) Another reason is that plans may exist even when the planning agent does not expect to achieve the specified objective. For example, a solo hobbiest wedding planner, who does not care whether the wedding plans come to fruition, still plans weddings, but does not expect that they will achieve some Objective.

2. In the definition of 'Intentional Act' the anaphoric reference by the phrase "the Agents" makes the definition read as either ungrammatical (because there is no reference to "the agents" in the first clause) or false (because it suggests that every intentional act requires more than one agent).

If you agree that the above should be addressed, I am happy to open a pull request and suggest changes.

@johnbeve

**Comment by PhiBabs935** [comment]
Regarding 1b), wouldn't that depend whether the Agent in the definition of plan is restricted to the person that made the plan?
"A Directive Information Content Entity that prescribes some set of intended Intentional Acts through which some Agent expects to achieve some Objective."

The way that I read the definition, the plan, created by the wedding planner, prescribes some intentional acts for the wedding participants to undertake. But I don't necessarily think the wedding planner has to be the one to expect to achieve the relevant Objective. Why wouldn't it be enough for the bride or groom to be the Agent that expects to achieve the relevant Objective by undertaking the intentional acts prescribed by the planner?

-Shane

**Comment by avsculley** [comment]
Hi Shane,

Thanks for the reply. My intention was for the hobbiest wedding planner to be the only actual agent in the case. Suppose that the hobbiest wedding planner does not intend that some wedding plan ever come to fruition (this is a slight variation to the initial case posted, but clarifies my intention). As such, if wedding participants exist at all in this case, it seems they exist as possible agents. Can cco:plan account for this? What is the sense in which a possible agent can have an expectation?

-Alec

**Comment by oliviahobai** [comment]
Moving issue to discussion thread so others can contribute
